File: 0605-can-place-flowers/0605-can-place-flowers.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        int i = 0;
        int length = flowerbed.Length;
        while(i < length){
            int prev = (i == 0) ? 0 : flowerbed[i - 1];
            int next = (i == length-1) ? 0 : flowerbed[i + 1];
            if (prev == 0 && flowerbed[i] == 0 && next == 0){
                flowerbed[i] = i;
                n--;
                if (n <= 0) return true;
                i += 2;
            } else{
                i++;
            }
        }
        return n <= 0;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.ts
function canPlaceFlowers(flowerbed: number[], n: number): boolean {
    let i: number = 0;
    let length: number = flowerbed.length;
    while (i < length) {
        let prev: number = (i === 0) ? 0 : flowerbed[i - 1];
        let next: number = (i === length - 1) ? 0 : flowerbed[i + 1];

        if (prev === 0 && flowerbed[i] === 0 && next === 0) {
            flowerbed[i] = 1;
            n--;
            if (n <= 0) return true;
            i += 2;
        } else {
            i++;
        }
    }
    return n <= 0;
};